### ISSUE TITLE
Running Lesson 7.1

### ISSUE THREAD

**Issue by johnthad** [hojberg/mastering-react#25]
Two simple changes will allow this lesson to run:
1. The `public` directory is missing. You must make the directory. Leave it empty.
2. Edit `package.json`, changing line #31 to force loading `react-router 1.0.0-beta3` _and only that version_: `"react-router": "1.0.0-beta3"`.

I haven't found the magic for handling server rendering and routing with the newest version of `react-router`. It seems to have [changed drastically](https://github.com/rackt/react-router/blob/master/docs/guides/advanced/ServerRendering.md). Maybe it can work with the `Router.run()` call in `server.js` (line #23) but `Location` no longer exists and I cannot get `history/lib/createLocation` to work. In any case, `react-router` [locations may be changing](https://github.com/rackt/react-router/issues/2186) and may well work some other way before it is finalized.

**Comment by hojberg** [comment]
Yeah I might have to re-record that lesson. The various beta and rc versions react-router changed dramatically.

**Comment by intrepion** [comment]
Instead of re-recording lessons, another option could be to make errata pages for each lesson and then add the same short recording to each video showing them the link to the errata pages.  On the errata pages, you can then post what exact versions that you need for the tutorial to work and, if you have time, post solutions for the latest versions.

**Comment by johnthad** [comment]
I don't think re-recording is necessary. In any case the stuff changes so
fast that whatever you did today might not apply in three months. Such is
the nature of bleeding edge.

I attended a local tech camp http://becamp.org/schedule yesterday where
one of the talks was on ReactJS. The presenter was from a local company
http://willowtreeapps.com/ that specializes in mobile apps. They are big
ReactJS cheerleaders. The presenter said they've had some of the same
issues getting to production.

You are welcome to post the file I sent you. Meanwhile I will try to
revisit #7.2, and see if I can make it work with the latest react-router.
As for #8 and #9, they were interesting, especially React Native. I did not
work closely though the demos because I have no current application for
them.

On Sun, Oct 25, 2015 at 11:07 AM, Oliver Forral notifications@github.com
wrote:

> Instead of re-recording lessons, another option could be to make errata
> pages for each lesson and then add the same short recording to each video
> showing them the link to the errata pages. On the errata pages, you can
> then post what exact versions that you need for the tutorial to work and,
> if you have time, post solutions for the latest versions.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/hojberg/mastering-react/issues/25#issuecomment-150933456
> .

## 

"Hell hath no limits, nor is circumscrib'd In one self-place; but where we
are is hell, And where hell is, there must we ever be" --Christopher
Marlowe, _Doctor Faustus_ (v. 121-24)
